staging/src/k8s.io/apiserver/pkg/storage/watch_cache.go

```diff
@@ -248,10 +251,10 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 	}
 	watchCacheEvent := &watchCacheEvent{
 		Type:                 event.Type,
-		Object:               event.Object,
-		ObjLabels:            objLabels,
-		ObjFields:            objFields,
-		ObjUninitialized:     objUninitialized,
+		Object:               elem.Object,
+		ObjLabels:            elem.Labels,
+		ObjFields:            elem.Fields,
+		ObjUninitialized:     elem.Uninitialized,
 		PrevObject:           prevObject,
```

Comment:
Seems like you can also get rid of `getAttrsFunc` computation for prevObject (which seems to be having all those fields present already now)?

Reply:
Good point.